build/cli.ts

```diff
@@ -220,7 +221,23 @@ async function buildDocuments(
     }
 
     for (const filePath of fileAttachments) {
-      // We *could* use symlinks instead. But, there's no point :)
+      // Ensure that binary files contain what their extension indicates.
+      if (/\.(mp3|mp4|ttf|webm|woff2?)$/i.test(filePath)) {
+        const ext = filePath.split(".").pop();
+        const type = await FileType.fromFile(filePath);
+        if (!type) {
+          throw new Error(
+            `Failed to detect type of file attachment: ${filePath}`
+          );
+        }
+        if (ext.toLowerCase() !== type.ext) {
+          throw new Error(
+            `Unexpected type '${type.mime}' (*.${ext}) detected for file attachment: ${filePath}.`
+          );
+        }
+      }
```

Comment:
TODO Move to filecheck.

Reply:
Limit to 500K

Reply:
> TODO Move to filecheck.

✅

Reply:
It would be good if this check could be file specific and based on the upper limit of current images of file types we use in MDN content and the interactive examples. 

I suspect 500K is fine for most cases, but perhaps not for mp3s/mp4s.

Will there be a way to choose to push the file through if you decide it is worth having despite the size?

Reply:
> Limit to 500K

✅ in https://github.com/mdn/yari/pull/7694

> I suspect 500K is fine for most cases, but perhaps not for mp3s/mp4s.

@hamishwillee On the one hand, mp3s/mp4s are easily/usually larger than that. But on the other hand, it's absolutely possible and wouldn't be bad to actually use/create mp3s/mp4s that are not larger than 500K (in terms of length/quality/resolution).

Reply:
For video/audio, if I have to spend time hunting around for files that fit the constraints (as well as the existing licensing constraints we have to worry about), then I'm much more likely just to use a remote URL that doesn't. Especially for examples which currently just work, like https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/preservesPitch#setting_the_preservespitch_property .

For fonts it is different because (1) they're not so big and (2) it's been impossible for us to find hosted fonts with the right features.

I think restricting sizes would be good *if* it came along with an asset library where we could collect assets that meet the restrictions. Which makes me feel that this PR, or part of it anyway, should perhaps be part of a bigger project. That way we only have to find an appropriate mp3 file once, rather than everyone having to go through it.

What do you think about:

1) landing this PR with a 500K restriction but only for existing file types and fonts. I think fonts are they only place this solves an immediate functional need (rather than, it's strategically nice to keep assets locally
2) create a repo where we can add assets meeting our criteria including video and audio
3) extend support for local files to audio and video
4) encourage people to use local assets instead of remote ones

?

Reply:
> For video/audio, if I have to spend time hunting around for files that fit the constraints (as well as the existing licensing constraints we have to worry about), then I'm much more likely just to use a remote URL that doesn't.

I would be curious why an audio/video file need to be larger than 500K in the context of our examples.

For example,
- 500K is enough to store 30 seconds of a 128kbps mp3,
- 500K is enough for 50 seconds of YouTube's 144p quality (80kbps), or 10 seconds of YouTube's 240p quality (377kbps).

Reply:
It's not that it is impossible, it's that it makes an extra barrier for writers.

We have a lot to do and don't want to spend more time than is needed finding example assets to use, as this is extra work beyond designing and writing the content.

For example when I [added a page for `HTMLMediaElement.preservesPitch`](https://github.com/mdn/content/pull/14183) I just looked for other places we were already using an mp3 in an example, and used the same one. Now we have a nice live sample in the page. But it's 1.3MB - if this rule were in place I'd now have to choose:

(1) use a local resource, and spend more time finding a different example
(2) use a remote resource and get the thing done now
(3) don't have a live sample

I'm pretty likely to choose (2). If we're next going to mandate local resources, maybe I'll choose (3) and the page will be worse. Or maybe in the original issue that the work came out of (https://github.com/mdn/content/issues/3783#issuecomment-1074513476) I'll think, oh, I don't have time to fix this now, I'll just add it to the pile of bugs.

Reply:
FWIW I'm happy with this as-is for now, with the chance that we have an assets library in future. It addresses the main use cases.

Generally we want it to be easy to add assets or people won't use it (as WillB says). But playing devil's advocate,  I personally don't want to have to sync large files in mdn/content because people didn't bother to crop their videos. Having an assets library makes this better, but can be a separate problem.

Can we get this in an leave the question of assets to a separate project?